website/source/docs/guides/kubernetes-reference.html.md

```diff
@@ -0,0 +1,184 @@
+--- 
+layout: "docs"
+page_title: "Kubernetes Consul Reference Architecture"
+sidebar_current: "docs-guides-k8s-reference-architecture"
+description: |-
+  This document provides recommended practices and a reference architecture. 
+---
+
+# Consul and Kubernetes Reference Architecture
+
+Preparing your Kubernetes cluster to successfully deploy and run Consul is an
+important first step in your production deployment process. In this guide you
+will prepare your Kubernetes cluster, independent of its underlying platform
+(AKS, EKS, GCP, etc). However, we will call out cloud specific differences when
+applicable. Before starting this guide you should have experience with
+Kubernetes, and have `kubectl` and helm configured locally. 
+
+By the end of this guide, you will be able to select the right resource limits
+for Consul pods, select the Consul datacenter design that meets your use case,
+and understand the minimum networking requirements. 
+
+## Infrastructure Requirements
+
+Consul server agents are responsible for the cluster state, responding to RPC
+queries, and processing all write operations. Since the Consul servers are
+highly active and are responsible for maintaining the cluster state, server
+sizing is critical for the overall performance efficiency and health of the
+Consul cluster. Review the [Consul Reference
+Architecture](/advanced/day-1-operations/reference-architecture#consul-servers)
+guide for sizing recommendations for small and large Consul datacenters. 
+
+The CPU and memory recommendations can be used when you select the resources
+limits for the Consul pods. The disk recommendations can also be used when
+selecting the resources limits and configuring persistent volumes. You will
+need to set both `limits` and `required` in the Helm chart. Below is an example
+Helm chart for a Consul server in a large environment.
+
+```yaml 
+server 
+  resources: | 
+    requests: 
+      memory: "32Gi" 
+      cpu: "4" 
+      disk: "50Gi"
+   limits: 
+     memory: "32Gi"
+     cpu: "4" 
+     disk: "50Gi" 
+```
+
+You should also set [resource limits for Consul
+clients](https://www.consul.io/docs/platform/k8s/helm.html#v-client-resources),
+so that the client pods do not unexpectedly consume more resources than
+expected. 
+
+[Persistent
+volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) (PV)
+allow you to have a fixed disk location for the Consul data. This ensures that
+if a Consul server is lost, the data will not be lost. This is an important
+feature of Kubernetes, but may take some additional configuration. If you are
+running Kubernetes on one of the major cloud platforms, persistent volumes
+should already be configured; be sure to read their documentation for more
```

Comment:
"should already be configured" seems ambiguous. Do you mean _the operator_ should have already configured them, or that the cloud provider should have done it magically (that's new to me) or something else like the Helm chart will do it automatically later?

Reply:
Most managed Kubernetes services already have a default storage class enabled for persistent volumes. Usually this takes advantage of their cloud storage option.

For example, "GKE creates a default StorageClass for you which uses the standard persistent disk type" from https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes